clap_complete/src/shells/fish.rs

```diff
@@ -20,8 +20,32 @@ impl Generator for Fish {
             .get_bin_name()
             .expect("crate::generate should have set the bin_name");
 
+        let has_global_flags = cmd.get_arguments().any(|a| {
+            !a.is_positional()
+                && a.get_long()
+                    .map(|name| !["help", "version"].contains(&name))
```

Comment:
I feel like this will need a comment

Reply:
By default, `-h/--help` and `-V/--version` are flags of a command. But they behaves as subcommands.
So here, I wanna exclude those flag to get the really useful flags that modifies behaviors of subcommands.

Reply:
That (and the comment) still aren't making sense to me.

Reply:
Ok, so if I'm understanding correctly, you are paying attention to `--help` and `--version` because you know that nothing after them matters, we short circuit argument parsing.

So if a parent command only has `--help` and `--version`,  you are assuming that  they won't appear before a subcommand (since those are usually "terminating" arguments) and so can ignore them and use a better way of parsing subcommands.

 Is that right?

Reply:
Yeah. it's something like that. Although, I wouldn't say the fallback is a better way of parsing subcommands.
It's just that I try to minimize the blessed changes on those `*.fish` files if the root command only has `--help` and `--version` flag.

Reply:
We shouldn't be doing things "just for the tests".

Reply:
Alright, I was worrying too much about the blessed completions changed from 10 lines to over 30 lines.
I'll remove `has_global_flags` as you prefer so.